documentation/docs/30-advanced/10-advanced-routing.md

```diff
@@ -123,27 +123,41 @@ src/routes/[...catchall]/+page.svelte
 
 ### Encoding
 
-Directory names are URI-decoded, meaning that (for example) a directory like `%40[username]` would match characters beginning with `@`:
+Some characters can't be used on the filesystem — `/` on Linux and Mac, `\ / : * ? " < > |` on Windows. The `#` character has special meaning in URLs, and the `[ ] ( )` characters have special meaning to SvelteKit, so these also can't be used directly as part of your route.
+
+To use these characters in your routes, you can use hexadecimal escape sequences, which have the format `[x+nn]` where `nn` is a hexadecimal character code:
+
+- `\` — `[x+5c]`
+- `/` — `[x+2f]`
+- `:` — `[x+3a]`
+- `*` — `[x+2a]`
+- `?` — `[x+3f]`
+- `"` — `[x+22]`
+- `<` — `[x+3c]`
+- `>` — `[x+3e]`
+- `|` — `[x+7c]`
+- `#` — `[x+23]`
+- `[` — `[x+5b]`
+- `]` — `[x+5d]`
+- `(` — `[x+28]`
+- `)` — `[x+29]`
+
+For example, to create a `/smileys/:-)` route, you would create a `src/routes/smileys/[x+3a]-[x+29]/+page.svelte` file.
+
+You can determine the hexadecimal code for a character with JavaScript:
 
 ```js
-// @filename: ambient.d.ts
-declare global {
-	const assert: {
-		equal: (a: any, b: any) => boolean;
-	};
-}
+':'.charCodeAt(0).toString(16); // '3a', hence '[x+3a]'
+```
 
-export {};
+You can also use Unicode escape sequences. Generally you won't need to as you can use the unencoded character directly, but if — for some reason — you can't use emoji in filenames, for example, then you can use the escaped characters. In other words, these are equivalent:
```

Comment:
The wording of this tripped me up for a second. "Emoji" is probably an _acceptable_ plural of "emoji" but I don't think that's universal enough to phrase this this way.

Reply:
Huh. Wikipedia says this...

> An emoji ([/ɪˈmoʊdʒiː/](https://en.wikipedia.org/wiki/Help:IPA/English) [i-MOH-jee](https://en.wikipedia.org/wiki/Help:Pronunciation_respelling_key); plural emoji or emojis[[1]](https://en.wikipedia.org/wiki/Emoji#cite_note-1)) is a [pictogram](https://en.wikipedia.org/wiki/Pictogram), [logogram](https://en.wikipedia.org/wiki/Logogram), [ideogram](https://en.wikipedia.org/wiki/Ideogram) or [smiley](https://en.wikipedia.org/wiki/Smiley) embedded in text and used in electronic messages and [web pages](https://en.wikipedia.org/wiki/Web_page). The primary function of **emoji** is to fill in emotional cues otherwise missing from typed conversation.[[2]](https://en.wikipedia.org/wiki/Emoji#cite_note-2) Examples of **emoji** are 😂, 😃, 🧘🏻‍♂️, 🌍, 🌦️, 🍞, 🚗, 📞, 🎉, ❤️, 🍆, 🍑 and 🏁. **Emoji** exist in various genres, including facial expressions, common objects, places and types of weather, and animals.

...but Buzzfeed says this:

<img width="620" alt="image" src="https://user-images.githubusercontent.com/1162160/201778813-210f6f1d-7d9a-40b1-aee2-ce74c3e8839e.png">

I guess this is just one of those cases where I will just have to clench my fist, smile, and pretend I don't know any better, like when people order 'a cannoli' or 'a panini'

Reply:
You could rework the sentence to skirt the first-clenching. "but if - for some reason - you can't use a filename with an emoji in it" or something similar.

Reply:
Also is your implication here that people should be saying "an emojo"